### PR TITLE
Feat/sanitize api

### PR DESCRIPTION
**Pull request by amimart**
Mainly improve the GoDoc of the provided APIs, along with some little refactoring for the sake of extensibility and consistency.

### CODE REVIEW

File: auth/jwt/claims.go [hunk 1]
```diff
@@ -2,11 +2,13 @@ package jwt
...
+// ProxyClaims is the set of claims that are included in the JWT token.
 type ProxyClaims struct {
 	jwt.StandardClaims
 	Can Permissions `json:"can"`
 }
 
+// Permissions is the set of permissions that are included in the JWT token.
 type Permissions struct {
 	Actions []string `json:"actions"`
 }
```
[the remark this answers is not in the document]
Comment (amimart) [reply auth/jwt/claims.go:14]: This is intentionally unrestrictive to match any needs, someone could through its governance rules express arbitrary actions for his own needs.

File: auth/jwt/jwt.go [hunk 2]
```diff
@@ -9,21 +10,24 @@ import (
...
-func NewFactory(secretKey []byte, issuer string, ttl time.Duration) *Factory {
-	return &Factory{
+// NewFactory creates a new Issuer with the given secret key, issuer and token time-to-live.
+func NewFactory(secretKey []byte, issuer string, ttl time.Duration) *Issuer {
+	return &Issuer{
 		secretKey: secretKey,
 		issuer:    issuer,
 		ttl:       ttl,
 	}
 }
```
[the remark this answers is not in the document]
Comment (amimart) [reply auth/jwt/jwt.go:27]: Oh you're right good catch! I'm a quite ashamed of myself.. 👀

**Review by ccamel — APPROVED** [review]
Great! 👍 Thx

**Review by bdeneux — COMMENTED** [review]
Thanks 🙏

**Pull request merged**